src/app/(main)/layout.tsx

```diff
@@ -1,20 +1,58 @@
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+"use client";
+
+import { useEffect } from "react";
+import { formatInTimeZone } from "date-fns-tz";
+import { useRouter } from "next/navigation";
 import Sidebar from "@/components/sidebar/Sidebar";
 import Navbar from "@/components/navbar/Navbar";
 import ModeToggle from "@/components/ModeToggle";
 import AuthHeader from "@/components/navbar/AuthHeader";
 import { getUser } from "@/utils/getUser";
-import AuthProvider from "@/app/(auth)/AuthProvider";
+import { useAppDispatch, useAuth } from "@/store/hooks";
+import { clientSignIn } from "@/store/features/auth/authSlice";
+import { currentDate } from "@/utils/getCurrentSprint";
+import { getUserState } from "@/store/features/user/userSlice";
+import routePaths from "@/utils/routePaths";
 
 interface LayoutProps {
   children: React.ReactNode;
 }
 
-export default async function Layout({ children }: LayoutProps) {
-  const [user, error] = await getUser();
+export default function Layout({ children }: LayoutProps) {
+  const dispatch = useAppDispatch();
+  const { isAuthenticated } = useAuth();
+  const router = useRouter();
+
+  useEffect(() => {
```

Comment:
Could we do something like that maybe?

// Function to fetch user and dispatch actions
const fetchUserAndDispatch = async () => {
  try {
    const data = await getUser();
    dispatch(clientSignIn());

    const currentDateInUserTimezone = formatInTimeZone(
      currentDate,
      data.timezone,
      "yyyy-MM-dd HH:mm:ss",
    );

    const userWithDate = {
      ...data,
      currentDate: new Date(currentDateInUserTimezone),
    };

    // Dispatch the getUserState action with the user object
    dispatch(getUserState(userWithDate));
  } catch (error) {
    console.error('Failed to fetch user', error);
    router.push(routePaths.signIn());
  }
};


  useEffect(() => {
    fetchUserAndDispatch(); // Call the separated function
  }, [dispatch, router]);